src/Microsoft.AspNetCore.Localization/QueryStringRequestCultureProvider.cs

```diff
@@ -71,9 +76,34 @@ public override Task<ProviderCultureResult> DetermineProviderCultureResult(HttpC
                 queryCulture = queryUICulture;
             }
 
+            if (!IsValidCulture(queryCulture))
+            {
+                _logger = _logger ?? httpContext.RequestServices.GetService<ILogger<QueryStringRequestCultureProvider>>();
+                _logger?.InvalidCultureName(nameof(QueryStringRequestCultureProvider), queryCulture);
+            }
+
+            if (!IsValidCulture(queryUICulture))
+            {
+                _logger = _logger ?? httpContext.RequestServices.GetService<ILogger<QueryStringRequestCultureProvider>>();
+                _logger?.InvalidCultureName(nameof(QueryStringRequestCultureProvider), queryUICulture);
+            }
+
             var providerResultCulture = new ProviderCultureResult(queryCulture, queryUICulture);
 
             return Task.FromResult(providerResultCulture);
         }
+
+        private static bool IsValidCulture(string cultureName)
```

Comment:
Can we move this to the super class instead of write it again in each implementation?